regression/goto-instrument/chain.sh

```diff
@@ -7,48 +7,33 @@ goto_instrument=$2
 cbmc=$3
 is_windows=$4
 
-name=${*:$#}
-
-if [[ x$name == x ]];  then
-  name=${name%.c}
-fi
-
+sources=${*:$#}
 args=${*:5:$#-5}
 
-if [[ "${is_windows}" == "true" && x$name != x ]]; then
-  $goto_cc "main.gb" ${name}
-  name="main"
-  mv "${name}.exe" "${name}.gb"
-elif [[ "${is_windows}" == "true" ]]; then
-  $goto_cc "${name}.c"
-  mv "${name}.exe" "${name}.gb"
-elif [[ x$name != x ]]; then
-  $goto_cc -o "main.gb" ${name}
-  echo "name: ${name}"
-  name="main"
+if [[ "${is_windows}" == "true" ]]; then
+  $goto_cc ${sources} "/Femain.gb"
 else
-  $goto_cc -o "${name}.gb" "${name}.c"
+  $goto_cc -o main.gb ${sources}
 fi
 
-rm -f "${name}-mod.gb"
-$goto_instrument ${args} "${name}.gb" "${name}-mod.gb"
-if [ ! -e "${name}-mod.gb" ] ; then
-  cp "$name.gb" "${name}-mod.gb"
+rm -f "main-mod.gb"
+$goto_instrument ${args} "main.gb" "main-mod.gb"
+if [ ! -e "main-mod.gb" ] ; then
+  cp "main.gb" "main-mod.gb"
 elif echo $args | grep -q -- "--dump-c-type-header" ; then
-  cat "${name}-mod.gb"
-  mv "${name}.gb" "${name}-mod.gb"
+  cat "main-mod.gb"
+  mv "main.gb" "main-mod.gb"
 elif echo $args | grep -q -- "--dump-c" ; then
-  cat "${name}-mod.gb"
-  mv "${name}-mod.gb" "${name}-mod.c"
+  cat "main-mod.gb"
+  mv "main-mod.gb" "main-mod.c"
 
   if [[ "${is_windows}" == "true" ]]; then
-    $goto_cc "${name}-mod.c"
-    mv "${name}-mod.exe" "${name}-mod.gb"
+    $goto_cc "main-mod.c" "/Femain-mod.gb"
   else
-    $goto_cc -o "${name}-mod.gb" "${name}-mod.c"
+    $goto_cc -o "main-mod.gb" "main-mod.c"
   fi
 
-  rm "${name}-mod.c"
+  rm "main-mod.c"
 fi
-$goto_instrument --show-goto-functions "${name}-mod.gb"
-$cbmc "${name}-mod.gb"
+$goto_instrument --show-goto-functions "main-mod.gb"
+$cbmc "main-mod.gb"
```

Comment:
The replacement of `${name}` by `main` here slightly concerns me as I know some of the regression tests can be run in parallel and this could conceivably cause a conflict or accidental overlap a some point. Has this been tested in such scenarios?

Reply:
That's a fair point, but then we already had this risk in here already for the code did `name=main` in each of the not-dead code branches. Let me try to come up with an actual improvement.

Reply:
That's now done. Thank you for calling this out!